### PR TITLE
Feature/stderr

### PR DESCRIPTION
**Pull request by ElvenSpellmaker**
- Adds comments to all class files.
- Allows `Error`s to be yielded which will write to the `stderr` of the system.

### CODE REVIEW

File: src/PipeSys/Command/DelayedHead.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?php
```
Comment (ElvenSpellmaker) [inline src/PipeSys/Command/DelayedHead.php:1]: This whole class, while useful in testing isn't very well thought out and isn't docblocked fully either. Change?

File: src/PipeSys/IO/Error.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	 * @var string
+	 */
+	protected $errorString;
+
+	public function __construct($errorString)
+	{
+		$this->errorString = (string)$errorString;
+	}
+
+	public function getErrorString()
```
Comment (ElvenSpellmaker) [inline src/PipeSys/IO/Error.php:20]: Docblock?

**Comment by ElvenSpellmaker** [comment]
Thinking about this some more tonight has made me realise, it'd be a lot better if commands could each have their own `StdErr`. Otherwise commands such as `yes 2> /dev/null | grep 'y'` isn't possible.

This would open a nice refactor which could potentially see the scheduler massively reduced in complexity for an abstract class which all commands extend which would contain most of the logic. Each command would then have its own `StdIn`, `StdOut

**Comment by ElvenSpellmaker** [comment]
Thinking about this some more tonight has made me realise, it'd be a lot better if commands could each have their own `StdErr`. Otherwise commands such as `yes 2> /dev/null | grep 'y'` aren't possible.

This would open a nice refactor which could potentially see the scheduler massively reduced in complexity for an abstract class which all commands extend which would contain most of the logic. Each command would then have its own `StdIn`, `StdOut`, and `StdErr`, with any that haven't been set being tied up by the scheduler at run-time.
Then each command could write directly to their StdIO channels without any care as they'll all be valid at run-time. (Relatively, they'd still need to yield, else you'd break the whole point of this coöperative scheduling.)

**Comment by JackBlower** [comment]
I finished said rewrite last night, will push later.

**Comment by JackBlower** [comment]
I finished said rewrite last night, will push later.

**Comment by ElvenSpellmaker** [comment]
I finished said rewrite last night, will push later.

**Pull request closed without merging**